### PR TITLE
[Moonshot/issues][#1027] Integrate multisig command

### PR DESCRIPTION
**Pull request by so-kkroy22**
## Description

- https://github.com/Entropy-Foundation/smr-moonshot/issues/1027
- https://github.com/Entropy-Foundation/smr-moonshot/pull/1196
- https://github.com/Entropy-Foundation/aptos-core/pull/109

## Change Kind / Issue Label
- [x] New Features

## Backward Compatibility

Compatible

## Target Release

smr-moonshot v6+

### CODE REVIEW

File: crates/aptos/src/account/multisig_account.rs [hunk 1]
```diff
@@ -372,3 +503,18 @@ impl CliCommand<TransactionSummary> for ExecuteReject {
...
+        let payload = aptos_stdlib::multisig_account_execute_rejected_transaction(
+            self.multisig_account.multisig_address,
+        );
+        Ok(SupraCommandArguments {
+            payload,
+            sender_account: self.txn_options.sender_account,
+            profile_options: supra_aptos::ProfileOptions::from(self.txn_options.profile_options),
+            rest_options: supra_aptos::RestOptions::from(self.txn_options.rest_options),
+            gas_options: supra_aptos::GasOptions::from(self.txn_options.gas_options),
+        })    }
```
Comment (isaacdoidge) [inline crates/aptos/src/account/multisig_account.rs]: Please make sure you have a formatter setup for this repo.

File: crates/aptos/src/account/multisig_account.rs [hunk 2]
```diff
@@ -345,6 +451,31 @@ impl CliCommand<TransactionSummary> for ExecuteWithPayload {
...
 }
 
+#[async_trait]
+impl SupraCommand for ExecuteWithPayload {
+    async fn supra_command_arguments(self) -> anyhow::Result<SupraCommandArguments> {
+        let payload = TransactionPayload::Multisig(Multisig {
+            multisig_address: self.execute.multisig_account.multisig_address,
+            transaction_payload: Some(self.entry_function_args.try_into()?),
+        });
+        Ok(SupraCommandArguments {
```
Comment (isaacdoidge) [inline crates/aptos/src/account/multisig_account.rs:480]: Why are these all dummy values except the payload?

File: crates/aptos/src/account/multisig_account.rs [hunk 3]
```diff
@@ -106,6 +107,29 @@ impl CliCommand<CreateSummary> for Create {
     }
 }
 
+#[async_trait]
+impl SupraCommand for Create {
+    async fn supra_command_arguments(self) -> anyhow::Result<SupraCommandArguments> {
+        let payload = aptos_stdlib::multisig_account_create_with_owners(
+            self.additional_owners,
+            self.num_signatures_required,
+            // TODO: Support passing in custom metadata.
```
Comment (isaacdoidge) [inline crates/aptos/src/account/multisig_account.rs]: Can you please add this as a part of this PR?

**Review by isaacdoidge — COMMENTED** [review]
A few questions. Also, do we still need to keep the old code around?

**Comment by so-kkroy22** [comment]
> A few questions. Also, do we still need to keep the old code around?

If the old code is not in use while developing the supra-framework then it can be removed.

**Review by isaacdoidge — APPROVED** [review]

**Pull request merged**